### ISSUE TITLE
Vscode 1.66.0: The Grammarly server crashed 5 times in the last 3 minutes. The server will not be restarted. 

### ISSUE THREAD

**Issue by universemaster** [znck/grammarly#230]
Possibly related to #214 .

This particular error is new in vscode 1.66.0.

```

/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38903
            throw new SocketError(this._statusCode, `Socket connection failed: ${this._statusMessage}`);
                  ^

SocketError: Socket connection failed: SHUTDOWN
    at GrammarlyClient._connect (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38903:19)
    at GrammarlyClient._queueOrSend (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38978:18)
    at /Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38876:18
    at new Promise (<anonymous>)
    at GrammarlyClient.send (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38874:16)
    at GrammarlyClient.sendWithErrorHandling (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39102:31)
    at GrammarlyClient.submitOT (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39194:21)
    at TextGrammarCheckHost.sync (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39943:24)
    at TextGrammarCheckHost.setText (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39886:19)
    at Function.update (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39686:28) {
  code: 1001
}
[Info  - 1:03:59 PM] Connection to server got closed. Server will restart.
1648814639698 DEBUG  [TextGrammarCheckHost] (66696c) Hosting file:///Users/me/Documents//a-level-further-math-a-proof.md
ResponseError: Unhandled method $/getToken
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563
                        responsePromise.reject(new messages_1.ResponseError(error.code, error.message, error.data));
                                               ^

ResponseError: Unhandled method $/updateDocumentState
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
[Info  - 1:03:59 PM] Connection to server got closed. Server will restart.
[Error - 1:03:59 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
    at Object.dispose (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:30704:25)
    at Object.dispose (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:31799:35)
    at LanguageClient.handleConnectionClosed (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:34034:42)
    at LanguageClient.handleConnectionClosed (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:35345:15)
    at closeHandler (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:34021:18)
    at CallbackList.invoke (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:29393:39)
    at Emitter.fire (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:29452:36)
    at closeHandler (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:30040:26)
    at CallbackList.invoke (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:29393:39)
    at Emitter.fire (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:29452:36)
    at IPCMessageReader.fireClose (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:30880:27)
    at ChildProcess.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/extension/index.js:30982:45)
    at ChildProcess.emit (node:events:402:35)
    at maybeClose (node:internal/child_process:1064:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:390:28)
    at Pipe.<anonymous> (node:net:687:12)
1648814639945 DEBUG  [TextGrammarCheckHost] (66696c) Hosting file:///Users/me/Documents/a-level-further-math-a-proof.md
ResponseError: Unhandled method $/getToken
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563
                        responsePromise.reject(new messages_1.ResponseError(error.code, error.message, error.data));
                                               ^

ResponseError: Unhandled method $/updateDocumentState
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
[Info  - 1:03:59 PM] Connection to server got closed. Server will restart.
1648814640157 DEBUG  [TextGrammarCheckHost] (66696c) Hosting file:///Users/me/Documents/a-level-further-math-a-proof.md
ResponseError: Unhandled method $/getToken
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563
                        responsePromise.reject(new messages_1.ResponseError(error.code, error.message, error.data));
                                               ^

ResponseError: Unhandled method $/updateDocumentState
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
[Info  - 1:04:00 PM] Connection to server got closed. Server will restart.
1648814640367 DEBUG  [TextGrammarCheckHost] (66696c) Hosting file:///Users/me/a-level-further-math-a-proof.md
ResponseError: Unhandled method $/getToken
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563
                        responsePromise.reject(new messages_1.ResponseError(error.code, error.message, error.data));
                                               ^

ResponseError: Unhandled method $/updateDocumentState
    at handleResponse (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28563:48)
    at processMessageQueue (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28390:17)
    at Immediate.<anonymous> (/Users/me/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:28374:13)
    at processImmediate (node:internal/timers:464:21) {
  code: -32601,
  data: undefined
}
[Error - 1:04:00 PM] Connection to server got closed. Server will not be restarted.
```

**Comment by learn2free** [comment]
same probl

**Comment by AlainOUYANG** [comment]
same here

**Comment by LiuX2018** [comment]
same problem.

**Comment by AlainOUYANG** [comment]
same here

**Comment by farzadso** [comment]
same here. disabling until further notice.

**Comment by kaeruchan** [comment]
same issue here.

**Comment by tristone13th** [comment]
same problem.

**Comment by rochdiboudjehem** [comment]
Downgrading to 1.65 solved the problem (https://code.visualstudio.com/updates/v1_65).
I also disabled automatic updates (for now)

N.B: to proceed I followed this link: https://stackoverflow.com/questions/49346733/how-to-downgrade-vscode

**Comment by codetroll** [comment]
Same issue here...

**Comment by Lightonz47** [comment]
Samme here

**Comment by AlainOUYANG** [comment]
Upgrading to Version: 1.66.1 seems to solve this problem.

**Comment by universemaster** [comment]
Same problem on Version: 1.66.1 for me. OS: Darwin x64 19.6.0.  MacOS Catalina, version 10.15.7.

**Comment by rochdiboudjehem** [comment]
> Upgrading to Version: 1.66.1 seems to solve this problem.

Me too

**Comment by rochdiboudjehem** [comment]
Version 1.66.1 worked at first, then begins to crash.

**Comment by tenhobi** [comment]
Version 1.66.1 (WSL) fixed this issue for me.

**Comment by tmsteen** [comment]
Sometimes, you may get a Connection to server got closed. Server will not be restarted. error. The best way to resolve this issue is to reload your VSCode window. And the easiest way to do that is with a keyboard shortcut. To set up the shortcut go to Code > Preferences > Keyboard Shortcuts. Then in the search box, search for workbench.action.reloadWindow. Double click on that item to set your keybinding. Option+Command+M seems to work well. 

When you reload the window, work is not lost but the extension will quickly start back up and operate as expected.

**Comment by elf-pavlik** [comment]
this is still happening in Version: 1.66.2 😭

**Comment by rangsimanketkaew** [comment]
Same for me, still experiencing this issue in VS Code 1.66.2.

**Comment by stemnic** [comment]
Following these instruction from an old issue and reloading the window works for me
VSCode 1.66.2 Macos 12.3.1 arm

> The `NOT_AUTHORIZED` error code occurs when access token received from OAuth has expired. I haven't implemented token deletion or refreshing. However, as a workaround you can:
> 
> 1. **Logout** using `Grammarly: Logout from grammarly.com`
> 2. **Reload**  or **Restart**  VS Code
> 3. **Login** using `Grammarly: Login to grammarly.com`
> 
> I don't have bandwidth to fix this issue right now but I am open to PRs.
> 
> _Originally posted by @znck in https://github.com/znck/grammarly/issues/183#issuecomment-879557150_

**Comment by zyrikby** [comment]
The same issue. The Logout->Reload->Login did not fix the issue.

VSCode version: 1.66.2
OS: Kubuntu 20.04

**Comment by universemaster** [comment]
**Symptom:** Often loading the Output panel in vscode is itself what seems to cause a crash.

**Possible solution:** Thus, I want to **turn off all logging**.

I have tried commenting out the `GrammarlyDiagnosticsService` from the package [grammarly-language-server](https://github.com/znck/grammarly/tree/e7f25fc8f675898e96ec9e2f9410eb0bed2148e8/packages/grammarly-language-server) at lines 8 and 35 [here](https://github.com/znck/grammarly/blob/e7f25fc8f675898e96ec9e2f9410eb0bed2148e8/packages/grammarly-language-server/src/index.ts#L8).

I am familiar with `vsce` but only have a little experience with `yarn`, and have been unable to build the extension, as I get errors.

I don't have the time to look into this further, but am leaving these thoughts here in case someone else wants to investigate.

**Comment by fangzhou-xie** [comment]
**Possible solution: Logout and don't login.**

I have experienced the same error on macOS VS Code but not on Fedora. Fedora was not logged in, but macOS did. So I wanted to try logout my macOS Grammarly plugin as well. As the plugin service was crashed and I couldn't logout, I uninstalled this plugin, reload window, install back, and I can use the service again!

I will keep using anonymous mode for the time being.

**Comment by theluke** [comment]
same here, Win 10 running on a MacBook Pro (on metal)

**Comment by sbwww** [comment]
> Following these instruction from an old issue and reloading the window worked for me VSCode 1.66.2 Macos 12.3.1 arm
> 
> > The `NOT_AUTHORIZED` error code occurs when access token received from OAuth has expired. I haven't implemented token deletion or refreshing. However, as a workaround you can:
> > 
> > 1. **Logout** using `Grammarly: Logout from grammarly.com`
> > 2. **Reload**  or **Restart**  VS Code
> > 3. **Login** using `Grammarly: Login to grammarly.com`
> > 
> > I don't have bandwidth to fix this issue right now but I am open to PRs.
> > _Originally posted by @znck in [#183 (comment)](https://github.com/znck/grammarly/issues/183#issuecomment-879557150)_

**Logout -> Reload -> Login -> Reload** works fine for me on VSCode 1.66.2, Win10

**Comment by cshjin** [comment]
Since Vscode 1.66 upgraded the Node.js from 14.16.0 to 16.13.0, there may cause node.js issues as we see. 
VScode [release note](https://code.visualstudio.com/updates/v1_66#_nodemoduleversion-and-nodejs-api-update)

Revert back to 1.65 or fix the problem with node.js should work.

**Comment by theluke** [comment]
`VsCode on Win 10 v 1.66.2`
Cannot logout as the client is not logged in, to begin with
Tried to restart the OS and restart the App
After about 5 min with the extension trying to log in, I receive the error  "Language Client is not available"

**Comment by xiaobaishu0097** [comment]
Same problem here.

VS Code Version: 1.66.2
OS: Darwin arm64 21.4.0

**Comment by theluke** [comment]
solved by disabling the plugin, restart, enable, login restart
But I dunno if the plugin ACTUALLY logs into Grammarly to benefit from my
business premium plan

**Comment by donangel** [comment]
Same problem (MacOS M1). Usually, I solve it by restarting VS Code. I hope there's a real solution soon.

**Comment by znck** [comment]
Fixed in extension v0.16

**Comment by ennioVisco** [comment]
> Fixed in extension v0.16

actually, I think the plugin broke again in v0.16 😅

**Comment by znck** [comment]
Probably a different error.
